# Exploratory Data Analysis
## Iowa Quarterly Retail Sales Tax Data

This notebook performs an initial exploratory data analysis (EDA) on the Iowa Quarterly Retail Sales Tax dataset from https://data.iowa.gov/Taxes-Tax-Credits/Iowa-Quarterly-Retail-Sales-Tax-Data-by-12-Busines/wsde-3id9/about_data

## 1. Load the Data

Import pandas and read the CSV file into a DataFrame called retail_data.

In [1]:
import pandas as pd

retail_data = pd.read_csv('Iowa_Quarterly_Retail_Sales_Tax_Data_by_12_Business_Groups_20260413.csv')

## 2. DataFrame Info

Use .info() to inspect column names, data types, and non-null counts. This helps identify missing values and whether columns need type conversion.

In [2]:
retail_data.info()

<class 'pandas.DataFrame'>
RangeIndex: 132586 entries, 0 to 132585
Data columns (total 10 columns):
 #   Column             Non-Null Count   Dtype
---  ------             --------------   -----
 0   Fiscal Year        132586 non-null  int64
 1   Quarter Ending     132586 non-null  str  
 2   County Number      132586 non-null  int64
 3   County             132586 non-null  str  
 4   Business Group     132585 non-null  str  
 5   Number of Returns  119622 non-null  str  
 6   Taxable Sales      119646 non-null  str  
 7   Computed Tax       119646 non-null  str  
 8   Percent of Tax     119646 non-null  str  
 9   Suppressed         132586 non-null  bool 
dtypes: bool(1), int64(2), str(7)
memory usage: 9.2 MB


Here we see discvepancies between the # of fiscal year to County (all 132586) and the rest of the datapoints, so we know there is missing data. 

## 3. First 5 Rows

Use .head() to preview the first 5 rows and get a feel for the structure, column names, and raw values (note currency and percent formatting in some columns).

In [3]:
retail_data.head()

,Fiscal Year,Quarter Ending,County Number,County,Business Group,Number of Returns,Taxable Sales,Computed Tax,Percent of Tax,Suppressed
0,2023,Sep 2022,42,Hardin,Apparel,21,"$357,947","$21,477",0.00%,False
1,2023,Sep 2022,45,Howard,Service,205,"$4,437,987","$265,848",0.04%,False
2,2023,Sep 2022,54,Keokuk,Wholesale,34,"$1,818,165","$109,090",0.01%,False
3,2023,Sep 2022,86,Tama,Wholesale,49,"$4,806,898","$288,414",0.04%,False
4,2001,Dec 2000,55,Kossuth,Motor Vehicle,31,"$2,821,069","$141,053",0.04%,False


## 4. Random Sample of 25 Rows

Use .sample(25) to view a random selection of rows across the dataset, giving a broader sense of the data distribution beyond just the first few records.

In [4]:
retail_data.sample(25)

,Fiscal Year,Quarter Ending,County Number,County,Business Group,Number of Returns,Taxable Sales,Computed Tax,Percent of Tax,Suppressed
54988,2009,Mar 2009,61,Madison,Motor Vehicle,17,"$701,718","$42,103",0.01%,False
64474,2011,Dec 2010,95,Winnebago,Eating and Drinking,31,"$1,443,498","$86,610",0.02%,False
132032,2026,Sep 2025,57,Linn,Home Furnishings,256,"$33,841,350","$2,030,481",0.26%,False
70451,2012,Mar 2012,58,Louisa,Specialty Retail,35,"$825,457","$49,527",0.01%,False
2535,2023,Dec 2022,91,Warren,Food Dealers,66,"$10,545,449","$632,727",0.08%,False
3164,2023,Mar 2023,39,Guthrie,County Totals,710,"$17,665,070","$1,059,109",0.17%,False
58234,2010,Dec 2009,11,Buena Vista,Miscellaneous,99,"$2,269,432","$136,166",0.03%,False
91851,2016,Jun 2016,16,Cedar,Apparel,NaN,NaN,NaN,NaN,True
61362,2010,Jun 2010,54,Keokuk,Building Materials,9,"$665,026","$39,902",0.01%,False
128430,2025,Dec 2024,75,Plymouth,Utilities and Transportation,87,"$5,664,577","$339,875",0.04%,False


Seems like missing datapoints are just null instead of various symbols etc. Moving forward I will remove these rows to clean the data.

In [8]:
retail_data_cleaned = retail_data.dropna()

print(f"Original rows : {len(retail_data)}")
print(f"Cleaned rows  : {len(retail_data_cleaned)}")
print(f"Rows removed  : {len(retail_data) - len(retail_data_cleaned)}")

#check if every row is the same length now
retail_data_cleaned.info()

Original rows : 132586
Cleaned rows  : 119611
Rows removed  : 12975
<class 'pandas.DataFrame'>
Index: 119611 entries, 0 to 132585
Data columns (total 10 columns):
 #   Column             Non-Null Count   Dtype
---  ------             --------------   -----
 0   Fiscal Year        119611 non-null  int64
 1   Quarter Ending     119611 non-null  str  
 2   County Number      119611 non-null  int64
 3   County             119611 non-null  str  
 4   Business Group     119611 non-null  str  
 5   Number of Returns  119611 non-null  str  
 6   Taxable Sales      119611 non-null  str  
 7   Computed Tax       119611 non-null  str  
 8   Percent of Tax     119611 non-null  str  
 9   Suppressed         119611 non-null  bool 
dtypes: bool(1), int64(2), str(7)
memory usage: 9.2 MB
